src/Raven.Client/Util/JavascriptConversionExtensions.cs

```diff
@@ -2671,6 +2671,10 @@ public override void ConvertToJavascript(JavascriptConversionContext context)
             }
         }
 
+        internal sealed class PatchDictionaryEnumSupport
```

Comment:
Why does it say about "dictionary"? How do we know that it will be applied only to dictionaries during patching operations?

Reply:
Because it's for generating the "jsonpath" for patches. I might be missing something but the only way to use enums in paths is when they are used as dictionary keys (`doc.Dict['EnumKey'] = 'value'`).